cpu/stm32f1/include/periph_cpu.h

```diff
@@ -59,6 +59,33 @@ typedef uint32_t gpio_t;
 #define TIMER_MAXVAL        (0xffff)
 
 /**
+ * @brief   Generate GPIO mode bitfields
+ *
+ * We use 4 bit to determine the pin functions:
+ * - bit 2+3: in/out
+ * - bit 1: PU enable
+ * - bit 2: OD enable
+ */
+#define GPIO_MODE(mode, cnf)    (mode | (cnf << 2))
+
+/**
+ * @brief   Override GPIO mode options
+ *
+ * We use 4 bit to encode CNF and MODE.
+ * @{
+ */
+#define HAVE_GPIO_MODE_T
+typedef enum {
+    GPIO_IN    = GPIO_MODE(0, 1),   /**< input w/o pull R */
+    GPIO_IN_PD = GPIO_MODE(0, 2),   /**< input with pull-down */
+    GPIO_IN_PU = GPIO_MODE(0, 2),   /**< input with pull-up */
```

Comment:
nope, this is correct -> this only enables the pull resistor in general, see `stm32f1/periph/gpio.c` line 89/90, the output register value determines if PU or PD are used.